src/tiled/mapeditor.cpp

```diff
@@ -314,6 +316,10 @@ void MapEditor::addDocument(Document *document)
         if (Layer *layer = layerAtGlobalIndex(mapDocument->map(), layerIndex))
             mapDocument->setCurrentLayer(layer);
     }
+
+    // Update all templates instances in this MapDocument when a template is edited
+    // Maybe it can be optimized
```

Comment:
Maybe!? :-)

It'd be cool to get it to the point where it is fast enough that we can apply rotation and size changes live instead of connecting to the index change of the undo history.